packages/components/src/popover/index.js

```diff
@@ -195,41 +195,39 @@ const Popover = (
 	 * Store the offset in a ref, due to constraints with floating-ui:
 	 * https://floating-ui.com/docs/react-dom#variables-inside-middleware-functions.
 	 */
-	const frameOffset = useRef( getFrameOffset( ownerDocument ) );
+	const frameOffsetRef = useRef();
 
 	const middleware = [
-		frameOffset.current || offset
```

Comment:
Removed the `frameOffset.current || offset` check, since it can cause a situation where offsetMiddleware is not added/removed when needed

Example:

- `frameOffset.current` starts as `undefined`
- therefore, `offsetMiddleware` is not added to the array
- `frameOffset.current` is assigned to an object (e.g `{x: 0, y: 32 }`)
- the change to `frameOffset.current` doesn't cause a re-render, and therefore  `offsetMiddleware` is still not added to the list of middlewares

Reply:
Thanks for catching that, this change makes sense 👍